R-package/src/lightgbm_R.cpp

```diff
@@ -56,8 +56,12 @@ LGBM_SE EncodeChar(LGBM_SE dest, const char* src, LGBM_SE buf_len, LGBM_SE actua
   return dest;
 }
 
-LGBM_SE LGBM_GetLastError_R(LGBM_SE buf_len, LGBM_SE actual_len, LGBM_SE err_msg) {
-  return EncodeChar(err_msg, LGBM_GetLastError(), buf_len, actual_len, std::strlen(LGBM_GetLastError()) + 1);
+SEXP LGBM_GetLastError_R() {
+  SEXP out;
+  out = PROTECT(Rf_allocVector(STRSXP, 1));
+  SET_STRING_ELT(out, 0, Rf_mkChar(LGBM_GetLastError()));
```

Comment:
Just a question: what is the difference between `Rf_allocVector`/`allocVector` and `Rf_mkChar`/`mkChar`?
I could find only the following link: https://www.cs.kent.ac.uk/projects/cxxr/doc/html/Rinternals_8h.html. And it looks like they are just aliases...

Reply:
yep, just aliases! This `#define` says "use the symbols with the `Rf_` prefix" https://github.com/microsoft/LightGBM/blob/1a367c65d182c0cc8d47e29e8f3d7a478dca3695/R-package/src/lightgbm_R.cpp#L15

I like that because it makes it a little more obvious which symbols are coming from R's headers.